Tests/RequestInterceptorTests.swift

```diff
@@ -312,6 +402,35 @@ final class InterceptorTestCase: BaseTestCase {
         XCTAssertEqual(result, .retry)
     }
 
+    func testThatInterceptorCanRetryRequestAsynchronously() {
+        // Given
+        let url = URL(string: "https://httpbin.org/get")!
+        let session = Session(startRequestsImmediately: false)
+        let request = session.request(url)
+
+        let retrier = Retrier { _, _, _, completion in
+            DispatchQueue.main.async {
+                completion(.retry)
+            }
+        }
+        let interceptor = Interceptor(retriers: [retrier])
+
+        var result: RetryResult!
+
+        let completesExpectation = expectation(description: "interceptor completes")
+
+        // When
+        interceptor.retry(request, for: session, dueTo: MockError()) {
+            result = $0
+            completesExpectation.fulfill()
+        }
+
+        _ = waitForExpectations(timeout: 0.01)
```

Comment:
Please use our default timeout. It should be available as `timeout`. Again, there doesn't seem to be a need for the assignment here.